cli/compose/convert/service.go

```diff
@@ -271,6 +271,10 @@ func convertServiceSecrets(
 			source = secretSpec.External.Name
 		}
 
+		if secretSpec.Name != "" {
+			source = namespace.Scope(secretSpec.Name)
+		}
```

Comment:
If the name is specified it shouldn't be scoped.

Reply:
ya, I think it's by design that the `name` field is not scoped. Only the default names get scoped.

You can use a variable in the `name:` field to provide the scope to these explicitly named objects.